### PR TITLE
BUG: fix optional support in decode_parameters

### PR DESCRIPTION
**Pull request by ebolyen**
Discovered while updated q2studio.

This should *probably* have unit-tests

### CODE REVIEW

File: qiime2/core/type/signature.py [hunk 1]
```diff
@@ -268,7 +268,10 @@ def _assert_valid_outputs(self, outputs):
     def decode_parameters(self, **kwargs):
         params = {}
         for key, spec in self.parameters.items():
-            params[key] = spec.qiime_type.decode(kwargs[key])
+            if spec.has_default() and spec.default is kwargs[key] is None:
```
Comment (jairideout) [inline qiime2/core/type/signature.py]: Woah, I've never seen this form of conditional before (`<something> is <something> is <something>`). I played around with it locally and it seems to work... maybe it's clearer to make this a more "standard" conditional to help with readability?

```python
if spec.has_default() and spec.default is None and kwargs[key] is None:
    ...
```

Comment (thermokarst) [inline qiime2/core/type/signature.py]: @ebolyen, I am with @jairideout on that one: can you update to his proposed form?

Comment (ebolyen) [inline qiime2/core/type/signature.py]: Sure, the method below works the same way FYI

**Comment by ebolyen** [comment]
For reference, here are [the docs](https://docs.paython.org/3/reference/expressions.html#comparisons) for why that chaining of `is` worked.

**Pull request merged**